cpp/src/strings/copying/copy_range.cu

```diff
@@ -0,0 +1,137 @@
+  CUDF_EXPECTS(
+    (target_begin >= 0) && (target_begin < target.size()) && (target_end <= target.size()),
+    "Range is out of bounds.");
+
+  if (target_end == target_begin) { return std::make_unique<column>(target.parent(), stream, mr); }
+  auto source_device_view = column_device_view::create(source.parent(), stream);
+  auto d_source           = *source_device_view;
+  auto target_device_view = column_device_view::create(target.parent(), stream);
+  auto d_target           = *target_device_view;
+
+  // create null mask
+  auto [null_mask, null_count] = cudf::detail::valid_if(
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(target.size()),
+    [d_source, d_target, source_begin, target_begin, target_end] __device__(size_type idx) {
+      return (idx >= target_begin && idx < target_end)
+               ? d_source.is_valid(source_begin + (idx - target_begin))
+               : d_target.is_valid(idx);
+    },
+    stream,
+    mr);
+
+  auto [check_source, check_target] = [target, null_count = null_count] {
+    // check validities for both source & target
+    if (target.has_nulls()) { return std::make_pair(true, true); }
+    // check validities for source only
+    if (null_count > 0) { return std::make_pair(true, false); }
+    // no need to check validities
+    return std::make_pair(false, false);
+  }();
+
+  // create offsets
+  auto sizes_begin = cudf::detail::make_counting_transform_iterator(
+    0,
+    compute_element_size{
+      d_source, d_target, source_begin, target_begin, target_end, check_source, check_target});
+  auto [offsets_column, chars_bytes] = cudf::strings::detail::make_offsets_child_column(
+    sizes_begin, sizes_begin + target.size(), stream, mr);
+  auto d_offsets = cudf::detail::offsetalator_factory::make_input_iterator(offsets_column->view());
+
+  // create chars
+  auto chars_data = rmm::device_uvector<char>(chars_bytes, stream, mr);
+  auto d_chars    = chars_data.data();
+  thrust::for_each(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator(0),
+    thrust::make_counting_iterator(target.size()),
+    [d_source, d_target, source_begin, target_begin, target_end, d_offsets, d_chars] __device__(
+      size_type idx) {
+      if (d_offsets[idx + 1] - d_offsets[idx] > 0) {
+        const auto source = (idx >= target_begin && idx < target_end)
+                              ? d_source.element<string_view>(source_begin + (idx - target_begin))
+                              : d_target.element<string_view>(idx);
+        memcpy(d_chars + d_offsets[idx], source.data(), source.size_bytes());
+      }
+    });
+
+  return make_strings_column(target.size(),
+                             std::move(offsets_column),
+                             chars_data.release(),
+                             null_count,
+                             null_count > 0 ? std::move(null_mask) : rmm::device_buffer{});
```

Comment:
Is it okay to make the column not nullable? the API description states that the effect is as if the copy was done in place.

Reply:
The test cases were failing since they are expecting non-nullable columns. 
I can change the tests to check for EQUIVALENT instead of EQUAL.

Reply:
is that is expected behavior, maybe just a note in the docs is sufficient.
I really don't know how/when this API is used, hence the question above.

Reply:
We should preserve the (non)nullability of the input, right?

Reply:
Right. That info was not available in the previous implementation but it certainly is now so we should be able to honor that here. I'll make sure the gtests do as well.